manifests/enterprise/repo/yum.pp

```diff
@@ -0,0 +1,33 @@
+# = Class: sensu::repo::yum
+#
+# Adds the Sensu YUM repo support
+#
+class sensu::enterprise::repo::yum {
+
+  if $caller_module_name != $module_name {
+    fail("Use of private class ${name} by ${caller_module_name}")
+  }
+
+  if $sensu::install_repo  {
+    if $sensu::enterprise {
+      $se_user = $sensu::enterprise_user
+      $se_pass = $sensu::enterprise_pass
+
+      validate_string($se_user, $se_pass)
+      if $se_user == undef or $se_pass == undef {
```

Comment:
What about moving this validation to init.pp?  Keeps all of the validation in one place then and when you get here you know you can just use the data then.